src/libAtomVM/ets_hashtable.c

```diff
@@ -94,38 +131,34 @@ EtsHashtableErrorCode ets_hashtable_insert(struct EtsHashTable *hash_table, term
 #endif
 
     struct HNode *node = hash_table->buckets[index];
+    struct HNode *last_node = NULL;
     if (node) {
```

Comment:
Not needed anymore since we are doing `while (node)`

Reply:
Consider adding a node with completely new key. We're using linked lists for collisions. If you don't have `last_node` then `node` will be always null – you'd create a new node and then instead of attaching it to the end of a linked list, you'd add it directly to the bucket, overwriting existing nodes.

Reply:
probably I didn't explain good enough, or maybe I'm overlooking something.
I mean that in the following snippet:
```c
if (cond) {
   while (cond) {
   }
}
```
`if (cond)` is redundant: if cond is false, while will be executed 0 times anyway.

Reply:
Nope, you're absolutely right. Github didn't show enough context in inline diff to see that.